commands/util/binary.js

```diff
@@ -0,0 +1,191 @@
+
+      if (subcommand === "encode") {
+        try {
+          const integer = interaction.options.getInteger("number");
+          const binary = integer.toString(2);
+
+          const embed = new EmbedBuilder()
+            .addFields(
+              { name: 'Number', value: `**${integer}**`, inline: true },
+              { name: 'Binary', value: `**${binary}**`, inline: true },
+            );
+
+          await interaction.reply({ embeds: [embed] });
+        } catch (error) {
+          await interaction.reply({ content: `An error occurred while encoding: ${error.message}`, ephemeral: true });
+        }
+
+      } else if (subcommand === "decode") {
+        try {
+          const binary = interaction.options.getInteger("binary");
+          const check = !/^[01]+$/;
+          if (check.test(binary.toString())) {
+            throw new Error("Input must be a valid binary number.");
+          }
+          const integer = parseInt(binary, 2);
+
+          const embed = new EmbedBuilder()
+            .addFields(
+              { name: 'Binary', value: `**${binary}**`, inline: true },
+              { name: 'Number', value: `**${integer}**`, inline: true },
+            );
+
+          await interaction.reply({ embeds: [embed] });
+        } catch (error) {
+          await interaction.reply({ content: `An error occurred while decoding: ${error.message}`, ephemeral: true });
+        }
+
+      } else if (subcommand === "add") {
+        try {
+          const first = interaction.options.getInteger("first");
+          const second = interaction.options.getInteger("second");
+          const check = !/^[01]+$/;
+          if (check.test(first.toString()) || check.test(second.toString())) {
+            throw new Error("Both inputs must be valid binary numbers.");
+          }
+          const sum = parseInt(first, 2) + parseInt(second, 2);
+          const binarySum = sum.toString(2);
+
+          const embed = new EmbedBuilder()
+            .addFields(
+              { name: 'First + Second', value: `First: **${first}**, Second: **${second}**`, inline: true },
+              { name: 'Sum', value: `**${binarySum}**`, inline: true },
+            );
+
+          await interaction.reply({ embeds: [embed] });
+        } catch (error) {
+          await interaction.reply({ content: `An error occurred while adding: ${error.message}`, ephemeral: true });
+        }
+
+      } else if (subcommand === "subtract") {
+        try {
+          const first = interaction.options.getInteger("first");
+          const second = interaction.options.getInteger("second");
+          const check = !/^[01]+$/;
+          if (check.test(first.toString()) || check.test(second.toString())) {
+            throw new Error("Both inputs must be valid binary numbers.");
+          }
+          const difference = parseInt(first, 2) - parseInt(second, 2);
+          const binaryDifference = difference.toString(2);
```

Comment:
What happens if the binaryDifference is negative? Does it still work? Not sure but was wondering if you tested it out.

Reply:
negative numbers don't work rn just positive numbers

Reply:
Yeah but what does it give you? As the output. There aren't any checks in place theree

Reply:
no it works

Reply:
i just tested it

Reply:
I went to a website and the negative version of a binary number is just -and then the number in binary